app/src/main/java/ch/epfl/sdp/appart/StepCounterActivity.java

```diff
@@ -0,0 +1,295 @@
+ * with the STEP_COUNTER and the STEP_DETECTOR sensor.
+ */
+public class StepCounterActivity extends AppCompatActivity implements SensorEventListener  {
+
+    /* UI components */
+    private TextView textViewStepCounter;
+    private TextView textViewStepDetector;
+    private TextView textViewKm;
+    private Button startButton;
+    private Button stopButton;
+    private ProgressBar progressBar;
+
+    /* sensors */
+    private SensorManager sensorManager;
+    private Sensor mStepCounter;
+    private Sensor mStepDetector;
+
+    private final static int STEP_COUNTER_PERMISSION_CODE = 110;
+
+    /* the below constants are for the 'number of meters' text view */
+    private final static double AVERAGE_STEP_SIZE_IN_METERS = 0.65;
+    private final static String METERS_UNIT = " meters";
+
+    /* in case this is not available (depends on the device running the app) the steps will be
+     * computed as a subtraction of current total STEP_COUNT with the previous total STEP_COUNT */
+    private static boolean stepDetectorSensorIsAvailable = false;
+
+    /* boolean values which determine the state of the StepCounter activity */
+    private static boolean startWasPressed = false;
+    private static boolean stopWasPressed = false;
+
+    /* this value is updated with the STEP_COUNTER sensor */
+    private static int totalStepCountFromBoot = 0;
+    /* these values are updated with the STEP_DETECTOR sensor */
+    private static int detectedStepsCount = 0;
+
+    /* number of steps registered by STEP_COUNT on start button pressed */
+    private static int initialTotalStepCountFromBoot = 0;
+    private static boolean initialTotalStepCountWasSet = false;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_step_counter);
+
+        textViewStepCounter = findViewById(R.id.totalNumberOfSteps_StepCounter_TextView);
+        textViewStepDetector = findViewById(R.id.numberOfSteps_StepCounter_TextView);
+        textViewKm = findViewById(R.id.km_StepCounter_TextView);
+        startButton = findViewById(R.id.startStepCount_StepCounter_Button);
+        stopButton = findViewById(R.id.stopStepCount_StepCounter_Button);
+        progressBar = findViewById(R.id.progress_StepCounter_ProgressBar);
+        progressBar.setVisibility(View.GONE);
+        startButton.setVisibility(View.VISIBLE);
+
+        /* activity recognition permission request */
+        PermissionRequest.askForActivityRecognitionPermission(this, () -> {
+            Log.d("PERMISSION", "Activity recognition permission granted");
+        }, () -> {
+            Log.d("PERMISSION", "Activity recognition permission refused");
+            finish();
+        });
+
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+    }
+
+    @Override
+    public void onBackPressed() {
+        finish();
+    }
```

Comment:
We can remove this one, as we discussed during today's meeting

Reply:
Yep!